diff --git a/drivers/acpi/pci_root.c b/drivers/acpi/pci_root.c
index 84030804a..da787cbb7 100644
--- a/drivers/acpi/pci_root.c
+++ b/drivers/acpi/pci_root.c
@@ -609,8 +609,7 @@ static void negotiate_os_control(struct acpi_pci_root *root, int *no_aspm)
 			 * it's unsupported. Leave existing configuration
 			 * intact and prevent the OS from touching it.
 			 */
-			dev_info(&device->dev, "FADT indicates ASPM is unsupported, using BIOS configuration\n");
-			*no_aspm = 1;
+			dev_info(&device->dev, "proceed with ASPM anyway\n");
 		}
 	} else {
 		/*
@@ -740,9 +739,6 @@ static int acpi_pci_root_add(struct acpi_device *device,
 		goto remove_dmar;
 	}
 
-	if (no_aspm)
-		pcie_no_aspm();
-
 	pci_acpi_add_bus_pm_notifier(device);
 	device_set_wakeup_capable(root->bus->bridge, device->wakeup.flags.valid);
 
diff --git a/drivers/gpu/drm/i915/display/intel_dp.c b/drivers/gpu/drm/i915/display/intel_dp.c
index e0e4cb529..4df0571fd 100644
--- a/drivers/gpu/drm/i915/display/intel_dp.c
+++ b/drivers/gpu/drm/i915/display/intel_dp.c
@@ -116,7 +116,7 @@ bool intel_dp_is_edp(struct intel_dp *intel_dp)
 {
 	struct intel_digital_port *dig_port = dp_to_dig_port(intel_dp);
 
-	return dig_port->base.type == INTEL_OUTPUT_EDP;
+	return dig_port->base.type == INTEL_OUTPUT_EDP || strncmp(intel_dp->attached_connector->base.name, "DP-3", 4) == 0;
 }
 
 static void intel_dp_unset_edid(struct intel_dp *intel_dp);
@@ -5337,7 +5337,7 @@ static bool _intel_dp_is_port_edp(struct drm_i915_private *dev_priv,
 	if (DISPLAY_VER(dev_priv) < 5)
 		return false;
 
-	if (DISPLAY_VER(dev_priv) < 9 && port == PORT_A)
+	if (DISPLAY_VER(dev_priv) < 9 && port == PORT_D)
 		return true;
 
 	return devdata && intel_bios_encoder_supports_edp(devdata);
diff --git a/drivers/gpu/drm/i915/display/intel_lvds.c b/drivers/gpu/drm/i915/display/intel_lvds.c
index 3ace56979..bc4d2d311 100644
--- a/drivers/gpu/drm/i915/display/intel_lvds.c
+++ b/drivers/gpu/drm/i915/display/intel_lvds.c
@@ -700,6 +700,14 @@ static const struct dmi_system_id intel_no_lvds[] = {
 			DMI_MATCH(DMI_PRODUCT_NAME, "P845"),
 		},
 	},
+	{
+		.callback = intel_no_lvds_dmi_callback,
+		.ident = "ThinkPad X230 Mod",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "LENOVO"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "23255NC"),
+		},
+	},
 	{ }	/* Terminating entry */
 };
 
diff --git a/drivers/pci/pci-acpi.c b/drivers/pci/pci-acpi.c
index a05350a4e..5237ea595 100644
--- a/drivers/pci/pci-acpi.c
+++ b/drivers/pci/pci-acpi.c
@@ -1505,8 +1505,7 @@ static int __init acpi_pci_init(void)
 	}
 
 	if (acpi_gbl_FADT.boot_flags & ACPI_FADT_NO_ASPM) {
-		pr_info("ACPI FADT declares the system doesn't support PCIe ASPM, so disable it\n");
-		pcie_no_aspm();
+		pr_info("ACPI FADT declares the system doesn't support PCIe ASPM, but we keep it enabled\n");
 	}
 
 	if (acpi_pci_disabled)
